cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/services/RequestService.java

```diff
@@ -0,0 +1,97 @@
+ */
+
+package com.farao_community.farao.cse.import_runner.app.services;
+
+import com.farao_community.farao.cse.import_runner.app.util.GenericThreadLauncher;
+import com.farao_community.farao.cse.runner.api.JsonApiConverter;
+import com.farao_community.farao.cse.runner.api.exception.AbstractCseException;
+import com.farao_community.farao.cse.runner.api.exception.CseInternalException;
+import com.farao_community.farao.cse.runner.api.resource.CseRequest;
+import com.farao_community.farao.cse.runner.api.resource.CseResponse;
+import com.farao_community.farao.cse.runner.api.resource.ThreadLauncherResult;
+import com.farao_community.farao.gridcapa.task_manager.api.TaskStatus;
+import com.farao_community.farao.gridcapa.task_manager.api.TaskStatusUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class RequestService {
+    private static final String TASK_STATUS_UPDATE = "task-status-update";
+    private static final Logger LOGGER = LoggerFactory.getLogger(RequestService.class);
+    private final CseRunner cseServer;
+    private final Logger businessLogger;
+    private final JsonApiConverter jsonApiConverter = new JsonApiConverter();
+    private final StreamBridge streamBridge;
+
+    public RequestService(CseRunner cseServer, Logger businessLogger, StreamBridge streamBridge) {
+        this.cseServer = cseServer;
+        this.businessLogger = businessLogger;
+        this.streamBridge = streamBridge;
+    }
+
+    public byte[] launchCseRequest(byte[] req) {
+        byte[] result;
+        CseRequest cseRequest = jsonApiConverter.fromJsonMessage(req, CseRequest.class);
+        // propagate in logs MDC the task id as an extra field to be able to match microservices logs with calculation tasks.
+        // This should be done only once, as soon as the information to add in mdc is available.
+        MDC.put("gridcapa-task-id", cseRequest.getId());
+        try {
+            streamBridge.send(TASK_STATUS_UPDATE, new TaskStatusUpdate(UUID.fromString(cseRequest.getId()), TaskStatus.RUNNING));
+            LOGGER.info("Cse request received : {}", cseRequest);
+            GenericThreadLauncher<CseRunner, CseResponse> launcher = new GenericThreadLauncher<>(cseServer, cseRequest.getId(), cseRequest);
+            launcher.start();
+            ThreadLauncherResult<CseResponse> cseResponse = launcher.getResult();
+            if (cseResponse.hasError() && cseResponse.getException() != null) {
+                throw cseResponse.getException();
+            }
+            Optional<CseResponse> resp = cseResponse.getResult();
+            if (resp.isPresent() && !cseResponse.hasError()) {
+                result = sendCseResponse(resp.get());
+                businessLogger.info("Cse response sent: {}", resp.get());
```

Comment:
I think this one can remain LOGGER, the information displayed is not really useful for the operator

Reply:
back to previous LOGGER